### ISSUE TITLE
Steps fail to match as expected when using Scenario Outline

### ISSUE THREAD

**Issue by andybflynn** [bencompton/jest-cucumber#1]
### This Issue is a ###
Possible bug, but maybe just a request for clarification

### The problem ###
If I write a Scenario Outline that uses examples, the example placeholders (e.g. <Item>) cause issues when writing a matching regex in your step definition.

### Example ###
```
// features/demo.feature
Feature: Demo test

Scenario Outline: I can double a number
  Given I call the doubleANumber function with the number <Num>
  Then It should return the number <ReturnVal>

  Examples:

  | Num | ReturnVal |
  | 1   | 2         |
  | 2   | 4         |
  | 3   | 6         |
```

```
// steps/demo.steps.js
const jestCucumber = require('jest-cucumber');
const doubleANumber = require('../demo.js');

const { loadFeature, defineFeature } = jestCucumber;
const feature = loadFeature('../features/demo.feature'));

defineFeature(feature, test => {
  test('I can double a number', ({given, then}) => {
    let val;
    given(/I call the returnANumber function with the number (\d+)/, (num) => {
      console.log(num);
      val = doubleANumber(parseInt(num, 10));
    });
    then(/It should return the number (\d+)/, (returnVal) => {
      expect(val).toBe(parseInt(returnVal, 10));
    });
  });
});
```

```
// demo.js
function doubleANumber(num) {  
  return num*2;
}

module.exports = doubleANumber;
```

This throws the following error:
`Expected step #1 in scenario "I can double a number" to match "I call the doubleANumber function with the number <Num>"`

The issue starts with the line `given(/I call the doubleANumber function with the number (\d+)/, (num) => { ... });`. 
I thought this would work because I'm expecting the captured value from the regex to be a number. However, this fails to match.

If I change the regex to `/I call the doubleANumber function with the number (<Num>)/` then it matches and runs, but the captured value sent to the callback is `undefined`.

If I then change the regex again to a combination of the two: `/I call the doubleANumber function with the number (<Num>|\d)/` then this works. It's almost like two parses are happening and the regex needs to support both the placeholder and the expected value type.

You can work around this by replacing the group match with a generic match everything `(.*)` but this feels messy. Am I missing something, or Is this intended behaviour, and what is the recommended way of working?

\*Edit\*
I should have been clearer, but when I'm making the changes to the regex in the above replication steps, I'm making the change to both regexes in the `given` and `then` steps in order to get the code to run.

**Comment by bencompton** [comment]
Thanks for the bug report!

For scenario outlines, the validation currently happens against the original step text from the feature file instead of the step text replaced with the example values. As you point out, this means that it currently requires a generic regex that can match both the original steps as well as the steps with the replacement values.

Using `(.*)` is the simplest way to make it work until this is fixed, but I agree that `(.*)` is not usually considered a best practice.

**Comment by andybflynn** [comment]
Thanks for the clarification Ben. Good to know I wasn't doing anything stupid. What do you think might be the best solution to this? Will it require a change to the way in which the step definitions are validated?

**Comment by bencompton** [comment]
In the [step definition validation](https://github.com/bencompton/jest-cucumber/blob/master/src/validation/step-definition-validation.ts#L41), it checks the steps for normal scenarios and scenario outlines the same way, which is to iterate the `steps` property. However, scenario outlines also have a `scenarios` property that contains a scenario for each example in the table, and each scenario has steps that have [already been parsed](https://github.com/bencompton/jest-cucumber/blob/master/src/parsed-feature-loading.ts#L125) with the replacement values for its example in the table.

Therefore, in the step definition validation, I am imagining that in the case of a scenario outline, it could validate against the processed steps for the first example instead of the unprocessed steps on the scenario outline itself.

**Comment by bencompton** [comment]
Fixed in v1.0.3.
